src/main/scala/com/dwolla/cloudflare/DnsRecordClient.scala

```diff
@@ -57,6 +57,26 @@ class DnsRecordClient(executor: CloudflareApiExecutor)(implicit val ec: Executio
     }
   }
 
+  def getExistingDnsRecordWithContentFilter(name: String, contentPredicate: String ⇒ Boolean, recordType: Option[String] = None): Future[Option[IdentifiedDnsRecord]] = {
+    getZoneId(domainNameToZoneName(name)).flatMap { zoneId ⇒
+      val parameters = Seq(Option("name" → name), recordType.map("type" → _))
+        .collect {
+          case Some((key, value)) ⇒ s"$key=${URLEncoder.encode(value, "UTF-8")}"
+        }
+        .mkString("&")
+      val request: HttpGet = new HttpGet(s"https://api.cloudflare.com/client/v4/zones/$zoneId/dns_records?$parameters")
+
+      executor.fetch(request) { response ⇒
+        val records = (response \ "result").extract[Set[DnsRecordDTO]]
+        val filteredRecords = records.filter(r ⇒ contentPredicate(r.content))
+        if (filteredRecords.size > 1) throw MultipleCloudflareRecordsExistForDomainNameException(name, filteredRecords)
```

Comment:
If we're using a filter instead of looking for a specific record, could we let the user of the CF client determine whether this should be an error?

Reply:
Altered the method to return `Future[Set[Identified]]` instead.